rosidl_typesupport_connext_c/cmake/rosidl_typesupport_connext_c_generate_interfaces.cmake

```diff
@@ -0,0 +1,242 @@
+  COMMAND ${PYTHON_EXECUTABLE} ${rosidl_typesupport_connext_c_BIN}
+  --generator-arguments-file "${generator_arguments_file}"
+  --dds-interface-base-path "${_dds_idl_base_path}"
+  --idl-pp "${_idl_pp}"
+  DEPENDS ${target_dependencies} ${_dds_idl_files}
+  COMMENT "Generating C type support for RTI Connext (using '${_idl_pp}')"
+  VERBATIM
+)
+
+set(_target_suffix "__rosidl_typesupport_connext_c")
+
+if(NOT WIN32)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+endif()
+
+link_directories(${Connext_LIBRARY_DIRS})
+add_library(${rosidl_generate_interfaces_TARGET}${_target_suffix} SHARED
+  ${_generated_msg_files} ${_generated_external_msg_files} ${_generated_srv_files})
+ament_target_dependencies(${rosidl_generate_interfaces_TARGET}${_target_suffix}
+  "rmw"
+  "rosidl_typesupport_connext_cpp")
+if(WIN32)
+  target_compile_definitions(${rosidl_generate_interfaces_TARGET}${_target_suffix}
+    PRIVATE "ROSIDL_BUILDING_DLL")
+  target_compile_definitions(${rosidl_generate_interfaces_TARGET}${_target_suffix}
+    PRIVATE "ROSIDL_TYPESUPPORT_CONNEXT_C_BUILDING_DLL")
+  target_compile_definitions(${rosidl_generate_interfaces_TARGET}${_target_suffix}
+    PRIVATE "NDDS_USER_DLL_EXPORT_${PROJECT_NAME}")
+endif()
+if(NOT WIN32)
+  set(_target_compile_flags "-Wall -Wextra")
+else()
+  set(_target_compile_flags
+    "/W4"
+    "/wd4100"
+    "/wd4127"
+    "/wd4275"
+    "/wd4458"
+    "/wd4701"
+  )
+endif()
+string(REPLACE ";" " " _target_compile_flags "${_target_compile_flags}")
+set_target_properties(${rosidl_generate_interfaces_TARGET}${_target_suffix}
+  PROPERTIES COMPILE_FLAGS ${_target_compile_flags})
+target_include_directories(${rosidl_generate_interfaces_TARGET}${_target_suffix}
+  PUBLIC
+  ${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_c
+  ${CMAKE_CURRENT_BINARY_DIR}/rosidl_typesupport_connext_c
+  ${CMAKE_CURRENT_BINARY_DIR}/rosidl_typesupport_connext_cpp
+)
+foreach(_pkg_name ${rosidl_generate_interfaces_DEPENDENCY_PACKAGE_NAMES})
+  set(_msg_include_dir "${${_pkg_name}_DIR}/../../../include/${_pkg_name}/msg/dds_connext_c")
+  set(_srv_include_dir "${${_pkg_name}_DIR}/../../../include/${_pkg_name}/srv/dds_connext_c")
+  normalize_path(_msg_include_dir "${_msg_include_dir}")
+  normalize_path(_srv_include_dir "${_srv_include_dir}")
+  target_include_directories(${rosidl_generate_interfaces_TARGET}${_target_suffix}
+    PUBLIC
+    "${_msg_include_dir}"
+    "${_srv_include_dir}"
+  )
+  ament_target_dependencies(${rosidl_generate_interfaces_TARGET}${_target_suffix}
+    ${_pkg_name})
+endforeach()
+ament_target_dependencies(${rosidl_generate_interfaces_TARGET}${_target_suffix}
+  "Connext"
+  "rosidl_typesupport_connext_c")
+
+add_dependencies(
+  ${rosidl_generate_interfaces_TARGET}
+  ${rosidl_generate_interfaces_TARGET}${_target_suffix}
+)
+add_dependencies(
+  ${rosidl_generate_interfaces_TARGET}${_target_suffix}
+  ${rosidl_generate_interfaces_TARGET}__cpp
+)
```

Comment:
I think you may need to add another dependency on the DDS IDL file generation which is done in the C++ generator. The target is created here: https://github.com/ros2/rmw_connext/blob/master/rosidl_typesupport_connext_cpp/cmake/rosidl_typesupport_connext_cpp_generate_interfaces.cmake#L25